### PR TITLE
Render attachments in conversation threads and details

### PR DESCRIPTION
**Pull request by zephraph**
In the conversation thread:
image

In the details panel:
image

Note that image attachments don't show up in the attachments details view (by design)

### CODE REVIEW

File: src/v2/Apps/Conversation/Components/Details.tsx [hunk 1]
```diff
@@ -52,6 +54,13 @@ export const Details: FC<DetailsProps> = ({ conversation, ...props }) => {
   const item =
     conversation.items?.[0]?.item?.__typename !== "%other" &&
     conversation.items?.[0]?.item
+
+  const attachments = conversation.messagesConnection.edges
+    .map(({ node }) => node.attachments)
+    .filter(attachments => attachments.length > 0)
+    .reduce((previous, current) => previous.concat(current), [])
+    .filter(attachment => !attachment.contentType.includes("image"))
```
Comment (zephraph) [inline src/v2/Apps/Conversation/Components/Details.tsx:62]: Eventually it'd be nice to replace this (and the related messagesConnection query) to be pulled directly from the conversation. As is, attachments in the details view will only render if they've already been loaded w/ the messages

File: src/v2/Apps/Conversation/Components/Details.tsx [hunk 2]
```diff
@@ -52,6 +54,13 @@ export const Details: FC<DetailsProps> = ({ conversation, ...props }) => {
   const item =
     conversation.items?.[0]?.item?.__typename !== "%other" &&
     conversation.items?.[0]?.item
+
+  const attachments = conversation.messagesConnection.edges
+    .map(({ node }) => node.attachments)
+    .filter(attachments => attachments.length > 0)
+    .reduce((previous, current) => previous.concat(current), [])
```
Comment (sepans) [inline src/v2/Apps/Conversation/Components/Details.tsx:61]: Maybe I am missing something but is this `reduce` line needed? It seems it is iterating over an array and adding the items back to an empty array?

File: src/v2/Apps/Conversation/Components/Message.tsx [hunk 3]
```diff
@@ -1,42 +1,83 @@
...
-      <Box>
-        <Link href={item.downloadURL}>{item.fileName}</Link>
-      </Box>
+      <AttachmentLink href={attachment.downloadURL} target="_blank">
+        <AttachmentContainer
+          alignSelf={alignSelf}
+          background={color(bgColor)}
+          p={1}
+          mt={0.5}
+        >
```
Comment (sepans) [inline src/v2/Apps/Conversation/Components/Message.tsx:73]: #veryminor seems possible to move the `if/else` clause inside `AttachementContainer` to make the code a little bit dryer.

File: src/v2/Apps/Conversation/Components/Details.tsx (hunk 2 above)
Comment (zephraph) [reply src/v2/Apps/Conversation/Components/Details.tsx:61]: The `reduce` function is being called on an array of arrays. `concat` has the interesting behavior in that if the input is an array it spreads it into the array it's being concatenated to (thereby flattening the array). Essentially the reduce is just taking the array of arrays and making it a flat array.

File: src/v2/Apps/Conversation/Components/Message.tsx (hunk 3 above)
Comment (zephraph) [reply src/v2/Apps/Conversation/Components/Message.tsx:73]: That's a good catch!

**Pull request merged**